### ISSUE TITLE
Decompiler: native-int conversion of argument address renders invalid `(nint)(ref x)` at Full

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1749]
## Defect

The decompiler emits invalid `Full` C# for methods that convert an argument address to `nint`/native int. The printer renders the address expression as `(nint)(ref bufferId)`, which is not legal C# expression syntax (`CS1525: Invalid expression term 'ref'`).

This shows up in CoreLib EventSource payload helpers that take addresses of primitive arguments and store them in `EventSource.EventData.DataPointer`.

## Repro

Run the CoreLib validity boss on the current runtime:

```bash
dotnet build src/dotnet-inspect -c Release

dotnet run --project tools/DecompilerHarness -c Release -- \
  --validity-check --compile-cap 800 --max-examples 30
```

Observed output:

```text
COMPILE-CHECK over 38350 rendered methods (37985 Full, 365 Partial)

Syntactic validity (parse + statement legality - false-positive-free):
  Full malformed   : 14 (0.04% of Full) - the "claimed good but won't parse" set
  Partial malformed: 271 (74.25% of Partial) - expected: the diagnosed unsupported bits

Malformed examples (Full):
  System.Buffers.ArrayPoolEventSource::BufferRented
    CS1525: Invalid expression term 'ref'
  System.Buffers.ArrayPoolEventSource::BufferAllocated
    CS1525: Invalid expression term 'ref'
  System.Buffers.ArrayPoolEventSource::BufferDropped
    CS1525: Invalid expression term 'ref'
  System.Threading.Tasks.TplEventSource::TaskScheduled
    CS1525: Invalid expression term 'ref'
  System.Threading.Tasks.TplEventSource::TaskCompleted
    CS1525: Invalid expression term 'ref'
  System.Threading.Tasks.TplEventSource::TaskWaitBegin
    CS1525: Invalid expression term 'ref'
  System.Threading.Tasks.TplEventSource::AwaitTaskContinuationScheduled
    CS1525: Invalid expression term 'ref'
  System.Threading.Tasks.TplEventSource::TraceSynchronousWorkEnd
    CS1525: Invalid expression term 'ref'
  System.Diagnostics.Tracing.EventSource::WriteEvent
    CS1525: Invalid expression term 'ref'
```

A concrete emitted example from `System.Buffers.ArrayPoolEventSource::BufferRented`:

```csharp
internal void BufferRented(int bufferId, int bufferSize, int poolId, int bucketId)
{
    byte* __stackalloc = stackalloc byte[checked(unchecked((nuint)4) * sizeof(EventSource.EventData))];
    EventSource.EventData* payload = (EventSource.EventData*)__stackalloc;
    payload.Size = 4;
    payload.DataPointer = (nint)(ref bufferId);
    payload.Reserved = 0;
    ((EventSource.EventData*)((byte*)payload + sizeof(EventSource.EventData))).Size = 4;
    ((EventSource.EventData*)((byte*)payload + sizeof(EventSource.EventData))).DataPointer = (nint)(ref bufferSize);
    // ... same pattern for poolId and bucketId ...
    base.WriteEventCore(1, 4, payload);
}
```

Original IL for the first invalid assignment:

```text
IL_0013: ldloc.0
IL_0014: ldarga.s     1
IL_0016: conv.u
IL_0017: call         System.Diagnostics.Tracing.EventSource.EventData::set_DataPointer(nint)
```

## Expected

The decompiler should not claim `Full` while emitting `(nint)(ref x)`. It should either render a valid unsafe address spelling for `ldarga` + `conv.u` (for example an address-of form where legal) or degrade honestly below `Full` with a diagnostic if the exact address expression cannot be represented.

Duplicate search found no direct open issue for the `DataPointer` / `Invalid expression term 'ref'` shape; this is a targeted instance of the broader invalid-Full burndown (#1687).

**Comment by richlander** [comment]
Additional cross-runtime confirmation from MAI follow-up hunt: the same invalid `Full` shape reproduces on .NET 9.0.13 CoreLib, not just the local .NET 11 preview runtime.

Command shape:

```bash
dotnet run --project tools/DecompilerHarness -c Release -- \
  /home/rich/.vscode-server-insiders/data/User/globalStorage/ms-dotnettools.vscode-dotnet-runtime/.dotnet/9.0.13~x64~aspnetcore/shared/Microsoft.NETCore.App/9.0.13/System.Private.CoreLib.dll \
  --validity-check --max-examples 100 --compile-cap 100
```

Reported `Full malformed : 14`, including `System.Buffers.ArrayPoolEventSource::BufferRented`, `BufferAllocated`, `BufferDropped`, `System.Threading.Tasks.TplEventSource::TaskScheduled`, `TaskCompleted`, `TaskWaitBegin`, `AwaitTaskContinuationScheduled`, `TraceSynchronousWorkEnd`, and several `System.Diagnostics.Tracing.EventSource::WriteEvent` overloads, all with `CS1525: Invalid expression term 'ref'`.\n\nDumped `BufferRented` renders the same invalid value-position address conversion:\n\n```csharp\nV_0.DataPointer = (nint)(ref bufferId);\n```\n\nThis confirms the row is a runtime-family pattern around argument-address conversion (`LoadArgumentAddress`), adjacent to but distinct from existing local-address coverage.

**Comment by richlander** [comment]
Claiming this #1687 burndown row as runner. Scope: fix native-int conversions of argument addresses so Full output no longer renders invalid `(nint)(ref x)`, add focused coverage/evidence, and drive to PR/blocker/pivot.

**Comment by richlander** [comment]
Runner update: #1749 is now in review via #1786. Evidence on PR: focused address canaries, decompiler PR-gate subset, product build, and CoreLib validity check showing the tracked BufferRented shape now renders `(nint)(&bufferId)` instead of `(nint)(ref bufferId)`.

**Comment by richlander** [comment]
Runner update: #1749 / #1786 is Ready to merge. Branch is clean, CI is green, local validation and adversarial review summary are posted on the PR.

**Comment by richlander** [comment]
Runner update: #1749 / #1786 remains Ready to merge after final sync with latest main and rerun CI.
